lib/handler/url_check.go

```diff
@@ -17,44 +15,45 @@ func urlCheck(session *discordgo.Session, orgMsg *discordgo.MessageCreate) {
 	for _, url := range parsed {
 		found, channelid, messageid := db.SearchLog(orgMsg, &orgMsg.GuildID, &url)
 		if found {
-			session.MessageReactionAdd(orgMsg.ChannelID, orgMsg.ID, config.CurrentConfig.Duplicate.React)
-			session.ChannelMessageSendReply(orgMsg.ChannelID, config.CurrentConfig.Duplicate.Message+"\nhttps://discord.com/channels/"+orgMsg.GuildID+"/"+channelid+"/"+messageid, orgMsg.Reference())
+			go session.MessageReactionAdd(orgMsg.ChannelID, orgMsg.ID, config.CurrentConfig.Duplicate.React)
+			replyMessage := config.CurrentConfig.Duplicate.Message + "\nhttps://discord.com/channels/" + orgMsg.GuildID + "/" + channelid + "/" + messageid
+			msg, _ := session.ChannelMessageSendReply(orgMsg.ChannelID, replyMessage, orgMsg.Reference())
+			go session.MessageReactionAdd(msg.ChannelID, msg.ID, config.CurrentConfig.Duplicate.Delete)
 		} else {
 			db.AddLog(orgMsg, &orgMsg.GuildID, &url, &orgMsg.ChannelID, &orgMsg.ID)
 		}
 	}
 }
 
 var (
-	urlWithPathRegex = regexp.MustCompile(`https?://[\w+.:?#[\]@!$&'()~*,;=/%-]+`)
+	re = regexp.MustCompile(`https?://[\w+.:?#[\]@!$&'()~*,;=/%-]+`)
 )
 
 func parseMsg(origin string) []string {
-	result := []string{}
-	if strings.HasPrefix(origin, "< ") {
-		return result
+	if strings.HasPrefix(origin, "<") {
+		return []string{}
 	}
-url_loop:
-	for _, rawUrl := range urlWithPathRegex.FindAllString(origin, -1) {
-		rawUrl = strings.ToLower(rawUrl)
-		rawUrl = strings.ReplaceAll(rawUrl, "youtu.be/", "youtube.com/watch?v=")
-		Url, err := url.Parse(rawUrl)
-		if err != nil {
-			log.Panic("Invalid URL: ", rawUrl)
-		}
-		for _, v := range config.CurrentConfig.DomainBlacklist {
-			if strings.Contains(Url.Host, v) {
-				continue url_loop
+
+	results := re.FindAllString(origin, -1)
+
+	for i, result := range results {
```

Comment:
In this case, I think using "standard `for`" is better than using `range`.
Using both of `results[i]` and `result` is confused.

Reply:
I changed some code, please take a look.